Gems/ROS2/Code/Source/Manipulator/ManipulatorControllerComponent.cpp

```diff
@@ -0,0 +1,302 @@
+#include <ROS2/Manipulator/ManipulatorControllerComponent.h>
+#include <RobotImporter/URDFMetadataComponent.h>
+#include "ROS2/ROS2Bus.h"
+#include <AzCore/Serialization/EditContext.h>
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+#include <PhysX/Joint/PhysXJointRequestsBus.h>
+#include <Source/HingeJointComponent.h>
+#include <AzCore/std/functional.h>
+#include <ROS2/VehicleDynamics/DriveModels/PidConfiguration.h>
```

Comment:
@danielemorra98 Would you mind verifying that https://github.com/o3de/sig-simulation/issues/54 covers the open aspects of this discussion, so that we can close this before moving the PR forward?

Reply:
I'd rather keep it out from [o3de/sig-simulation#54](https://github.com/o3de/sig-simulation/issues/54) and open another PR just for this specific refactoring